packages/react-native-web/src/exports/Image/index.js

```diff
@@ -337,24 +345,94 @@ const Image: React.AbstractComponent<
   );
 });
 
-Image.displayName = 'Image';
+/**
+ * This component handles specifically loading an image source with header
+ */
+const ImageWithHeaders: ImageComponent = React.forwardRef((props, ref) => {
+  // $FlowIgnore
+  const nextSource: SourceObject = props.source;
+  const prevSource = React.useRef<SourceObject>({});
+  const cleanup = React.useRef<Function>(() => {});
+  const [blobUri, setBlobUri] = React.useState('');
+
+  const { onError, onLoadStart } = props;
+
+  React.useEffect(() => {
+    if (!hasSourceDiff(nextSource, prevSource.current)) return;
+
+    // When source changes we want to clean up any old/running requests
+    cleanup.current();
+
+    prevSource.current = nextSource;
+
+    let uri;
+    const abortCtrl = new AbortController();
+    const request = new Request(nextSource.uri, {
+      headers: nextSource.headers,
+      signal: abortCtrl.signal
+    });
+    request.headers.append('accept', 'image/*');
+
+    if (onLoadStart) onLoadStart();
+
+    fetch(request)
+      .then((response) => response.blob())
+      .then((blob) => {
+        uri = URL.createObjectURL(blob);
+        setBlobUri(uri);
+      })
+      .catch((error) => {
+        if (error.name !== 'AbortError' && onError) {
+          onError({ nativeEvent: error.message });
+        }
+      });
```

Comment:
Perhaps it would be best to move this logic to `ImageLoader.loadUsingHeaders` as it would be easier to mock inside tests 

Otherwise we have to mock
- Request
- fetch
- URL.createObjectURL and revokeObjectURL